### DISCUSSION TITLE
MLX90614 Console "mlx checksum error"

### DISCUSSION THREAD

**Discussion by Wolle62** [arendst/Tasmota#15821] [Q&A] [answered]
I have a question about an error in the Tasmota console.
I compiled a version with MLX90614 support.
Everything works fine, but in the console I see an error every now and then "mlx checksum error.
no pattern is discernible. Can I ignore the error? Or is the module no longer 100% compatible with the latest version of Tasmota?

**Comment by sfromis** [comment, marked as answer]
Checksum error would usually be a problem with transmitting the data from the sensor to the MCU running Tasmota. It could be issues with details of wiring, power or noise. Potentially also if the Tasmota driver has problems handling the data.

**Reply by Wolle62** [reply]
> It could be issues with details of wiring, power or noise.
The Wemos D1 Mini connects directly to a USB3 port on the PC.
The I2C wires are 3cm long and I reduced the pull-up resistors to 2.2k on the MLX module.
How often do data errors occur on the I2C bus in practice? Is that possibly normal every 10 minutes with a polling of 1s?
Then I could just ignore this message?

**Reply by sfromis** [reply]
Normal error rate is zero, unless there are issues, which is not merely the things you mention. Anyway, if you only feel the pain when looking at the console, it may not be too bad :stuck_out_tongue_closed_eyes:

**Reply by ghost** [reply]
EDIT: I Use the Addafruit MLX90614 2.1.3 Library and the MLXTEST Sketch

Hello, I have the same problem and I think it is the Library and not the wiring and not the Sensor itself. 
I say this because I also use the the Arduino Test sketch without any error. The sketch uses the hardware I2C wires of ESP8266 
D1SCL  D2SDA  So I gave TASMOTA a second try with this pins, but no luck. 
I may have to do with the link speed or some other critical timings but it is not the sensor or connectors.
Arduino Sketch Serial Monitor give me:
20:10:03.309 -> Ambient = 79.75*F	Object = 80.08*F
20:10:03.309 -> 
20:10:03.781 -> Ambient = 26.53*C	Object = 26.71*C
20:10:03.781 -> Ambient = 79.75*F	Object = 80.08*F
20:10:03.781 -> 
20:10:04.313 -> Ambient = 26.53*C	Object = 26.85*C
20:10:04.313 -> Ambient = 79.75*F	Object = 80.33*F
20:10:04.313 -> 
20:10:04.809 -> Ambient = 26.53*C	Object = 26.91*C
20:10:04.809 -> Ambient = 79.75*F	Object = 80.44*F

With Tasmota I get:
20:07:09.433 mlx checksum error
20:07:10.438 mlx checksum error
20:07:11.416 mlx checksum error
20:07:12.423 mlx checksum error
20:07:13.451 mlx checksum error
20:07:14.418 mlx checksum error
20:07:15.427 mlx checksum error
20:07:16.444 mlx checksum error
20:07:17.449 mlx checksum error
20:07:18.411 mlx checksum error
20:07:19.435 mlx checksum error
20:07:20.441 mlx checksum error
20:07:21.449 mlx checksum error
20:07:22.421 mlx checksum error
20:07:24.432 mlx checksum error
20:07:25.408 mlx checksum error
20:07:27.421 mlx checksum error
20:07:29.455 mlx checksum error
20:07:30.414 mlx checksum error
20:07:31.445 mlx checksum error

**Comment by Wolle62** [comment]
I taked the solder iron and a Wemos Shield. Now there are no breadbord or any other wiring problems.
The Error stays. Damn...
I hoped, here is anyone who used the MLX90614 to and can help with Expieriences.
May be a timing Problem?

**Reply by Wolle62** [reply]
Found this here:
the ESP8266(ESP32) activates its internal pull-up resistors for SDA and SCL, so you usually don't need to install external ones (they are required for slaves).

It means, that i could try to remove the resistors on the MLX Modul?
My Controller is a ESP8266...

EDIT: try it, the Error stays without Pull-Ups.

**Reply by sfromis** [reply]
It is very typical for I2C breakout boards to already have the pullup resistors needed for I2C, and I've not seen any suggestions of these hurting in any way. If the breakout board did not have them, adding pullups might be helpful in case of the weak internal ESP pullups not being strong enough.

If you "only" have an error every 10 minutes, and it works fine otherwise, you could also calibrate your pain threshold....

**Comment by Wolle62** [comment]
Without Pull-ups and Breadboard i have only 3-5  errors per hour. May be the HF Radiation of WLAN make this, but its ok now. 
Sorry for bad english and thank you for helping.

**Comment by Wolle62** [comment]
Ok, i changed the code from the "xsns_46_MLX90614.ino" in Tasmota from
 "FUNC_EVERY_SECOND" to "FUNC_AFTER_TELEPERIOD" in Line 151!
Now i get the temperature from the MLX90614 sensor only after every  teleperiod and not every second.
For a fishtank its enough.
It was the easiest way to get out this error and reduce the load.
The teleperiod ca be changed from the console, so its easy to mange it.
Maybe this Info is usefull for someone...
